repair/row_level.cc

```diff
@@ -396,45 +396,135 @@ using is_local_reader = bool_class<class is_local_reader_tag>;
     }
 };
 
+class repair_writer_impl : public repair_writer::impl {
```

Comment:
In the ideal world (outside of the scope of this series), repair_meta would receive a repair_writer as injected dependency instead of creating it internally. The patch series does NOT instantiate/test repair_meta, so there was no need for that change. That being said, the repair_writer used in test does use test_repair_writer_impl (which inherits from repair_writer::impland test_mutation_fragment_queue_impl (which inherits from mutation_fragment_queue::impl). 

repair_writer::impl was abstracted to hide the process of creating the underlying writer (multishard vs the test one) and waiting until it's finished writing. The queue itself is a separate concept as it exposes an api to write to it and hides the details of the consumer of that queue (writing to sstables in production code vs the test one).

In the pre-patch case you could test what the sstable has written to the disk, but you still need a channel to provide the relevant mutations. This patch series is written with the subsequent change in mind, which changes the internal representation of repair_row (which contains mutation_fragment) to repair_row_v2 (which contains mutation_fragment_v2). The whole path then takes repair_rows_on_wire (contains mutation_fragment v1) converts to internal representation and then flushes to disk. There is no point in testing the repair_row (or repair_row_v2) itself, as this is an implementation detail. We need to ensure the path of repair_rows_on_wire to disk is executed properly, regardless of the implementation details.